cm-ovirt-vm-creator.py

```diff
@@ -0,0 +1,193 @@
+            counter = 1
+            while counter < args.num_of_iterations:
+                time.sleep(args.sleep_between_iterations)
+                vm = vm_service.get()
+                print ("vm.status = %s"%(vm.status))
+                if vm.status == types.VmStatus.DOWN:
+                    break
+            vm_service.start( use_cloud_init=True, vm=types.Vm(initialization=types.Initialization(authorized_ssh_keys=os.environ[args.pub_sshkey])))
+            counter = 1
+            while counter < args.num_of_iterations:
+                time.sleep(args.sleep_between_iterations)
+                vm = vm_service.get()
+                print ("vm.status = %s, vm.fqdn= '%s'"%(vm.status, vm.fqdn))
+                if vm.status == types.VmStatus.UP  or counter > 20:
+                    break
+            time.sleep(args.sleep_between_iterations)
+    #print (vm_dict)
+
+
+def main():
+    # Parse command line arguments
+    parser = argparse.ArgumentParser(description='Creates a set of VMs to be used by cm-jenkins as  openshift nodes')
+    
+    #Mandatory Parameters
+    parser.add_argument('--name-prefix', type=str, required=True, help='The name to be used as a prefix for all the created VMs')
+    parser.add_argument('--ovirt-url', type=str, required=True, help='The url pointing to the oVirt Engine API end point')
+    parser.add_argument('--ovirt-user', type=str, required=True, help='The user to use to authenticate with the oVirt Engine')
+    parser.add_argument('--ovirt-ca-pem-file', type=str, required=True, help='Path to the ca pem file to use when connecting to tyhe engine')
+    parser.add_argument('--ovirt-cluster', type=str, required=True, help='The cluster name where to create the VMs on')
+    parser.add_argument('--ovirt-template', type=str, required=True, help='The Template to use for the VM creation')
+     
+    #optional arguments
+    parser.add_argument('--info', const=True, nargs='?', type=str2bool, default=False, help='Used to obtain all the VM ips')
+    parser.add_argument('--masters', const=1, nargs='?', type=int, default=1, help='Number of master nodes to create in the cluster')
+    parser.add_argument('--nodes', const=3, nargs='?', type=int, default=3, help='Number of compute nodes to create in the cluster')
+    parser.add_argument('--infra-nodes', const=2, nargs='?', type=int, default=2, help='Number of infra nodes to create in the cluster')
+    parser.add_argument('--ovirt-pass', const="OV_PASS", nargs='?', type=str, default="OV_PASS", help='The env variables to use as the Password to authenticate to oVirt')
+    parser.add_argument('--pub-sshkey', const="OV_SSH_KEY", nargs='?', type=str, default="OV_SSH_KEY", help='The env variables to use for the public ssh key to insert into the created VMs')
+    parser.add_argument('--num-of-iterations', const=20, nargs='?', type=int, default=20, help='Number of iterations to wait for long VM operations (creation & wait for up status)')
```

Comment:
Perhaps it'd be more user-friendly to replace `num-of-iterations` and `sleep-between-iterations` with a simple `timeout` parameter, have the sleep duration constant and adjust the number of retries according to the requested timeout?